launch/launch/actions/execute_process.py

```diff
@@ -555,7 +552,7 @@ def __flush_buffers(self, event, context):
 
         # the respawned process needs to reuse these StringIO resources,
         # close them only after receiving the shutdown
-        if self.__shutdown_received:
+        if self.__shutdown_future is None or self.__shutdown_future.done():
```

Comment:
why `or`? shouldn't be `and` here?

Reply:
@ivanpauno 
I think you mean **(self.__shutdown_future is not None and self.__shutdown_future.done())** here.
**self.__shutdown_future is None or self.__shutdown_future.done():** contains
**self.__shutdown_future is None** or **(self.__shutdown_future is not None and self.__shutdown_future.done())**.
Just for __flush_buffers() this interface, I think it needs to check if **self.__shutdown_future** is created or not.
So I use **or** here.

Reply:
After checking the code again, this looks correct to me.

Reply:
This is a peculiar way of writing this condition. `self.__shutdown_future is None` is a tautology in normal circumstances, as no buffers will ever be flushed if the process hasn't even started.

Reply:
We create the buffers right away in `__init__`:

https://github.com/ros2/launch/blob/0472f9a3ce6a156dfa93ccd0261b9727cadacc47/launch/launch/actions/execute_process.py#L247-L248

thus I think this check is correct

Reply:
Sure, my point is that if the process never started then `__flush_buffers()` will never be called.

Reply:
That's a good point.
We should probably close buffers in `__cleanup` (no related to this PR).